types/tx.go

```diff
@@ -42,11 +45,13 @@ func (tx Tx) String() string {
 // Txs is a slice of Tx.
 type Txs []Tx
 
-// Hash returns the Merkle root hash of the transaction hashes.
-// i.e. the leaves of the tree are the hashes of the txs.
+// Hash returns the Merkle root hash.
+// The leaves of the tree are the  hashes of the txs.
+//
+// See merkle.HashFromByteSlicesWithHash.
 func (txs Txs) Hash() []byte {
 	hl := txs.hashList()
-	return merkle.HashFromByteSlices(hl)
+	return merkle.HashFromByteSlicesWithHash(txhash.New(), hl)
```

Comment:
We break `Header#DataHash` here. I.e. another chain won't be able to compute this without the knowledge of the hashing algorithm used by this chain.